enhancements/shared-and-guaranteed-cpu-pools.md

```diff
@@ -0,0 +1,407 @@
+
+Here are the CPU requests/limits from an example container that uses shared CPUs:
+```yaml
+requests:
+  cpu: 200m
+limits:
+  cpu: 400m
+```
+This would be mutated as follows:
+```yaml
+requests:
+  cpu: 200m
+  openshift.io/shared-cpus: 200
+limits:
+  cpu: 400m
+  openshift.io/shared-cpus: 200
+```
+
+Kubelet's cgroup creation/management code will continue to use the CPU requests/limits to configure the CFS shares/quota
+for each container and will not require any modifications.
+
+### Workflow Description
+
+**cluster creator** is a human user responsible for deploying a cluster.
+
+1. The cluster creator creates a Performance Profile for the NTO and specifies a shared CPU partition.
+2. The cluster creator creates a machine config manifest to write a configuration file for kubelet to specify
+the shared and isolated CPUs (TODO: or this could be the config option).
+3. The cluster creator (TODO: does what???) to enable the new admission hook.
+4. The cluster creator then creates the cluster.
+5. The kubelet starts up, finds the configuration file and initializes the `defaultCpuSet` and `guaranteedCpuSet` 
+based on the shared/isolated CPUSets specified in the config file.
+6. The kubelet advertises `openshift.io/shared-cpus` and `openshift.io/guaranteed-cpus` extended resources 
+on the node based on the shared/isolated CPUSets specified in the config file.
+7. Something schedules:
+   * a pod with the `target.workload.openshift.io/management` annotation in a namespace 
+with the `workload.openshift.io/allowed` management annotation. The admission hook ignores the pod as it will
+be handled by the managementcpusoverride admission hook.
+   * a pod with Burstable or BestEffort QoS. The admission hook modifies the pod,
+adding `openshift.io/shared-cpus` requests/limits for each container, matching the CPU requests.
+   * a pod with Guaranteed QoS. The admission hook modifies the pod,
+adding `openshift.io/guaranteed-cpus` requests/limits for any containers with whole CPU requests/limits and
+`openshift.io/shared-cpus` requests/limits for other containers, matching the CPU requests.
+8. The scheduler sees the new pod and finds available `openshift.io/shared-cpus` and/or 
+`openshift.io/guaranteed-cpus` resources on a node. The scheduler places the pod on the node.
+9. Repeat steps 7-8 until all pods are running.
+
+### API Extensions
+
+A new admission hook in the Kubernetes API Server within OpenShift will mutate pods when they are created
+to add `openshift.io/guaranteed-cpus` and/or `openshift.io/shared-cpus` requests/limits as described above.
+
+### Implementation Details/Notes/Constraints
+
+One constraint of this approach is that the feature scope is cluster wide. When the feature gate is
+enabled, the new admission hook will mutate all incoming pods, adding requests/limits for the new
+extended resources. This means that all nodes in the cluster must have a Performance Profile that
```

Comment:
This is a very strict requirements.
This means that https://github.com/openshift/enhancements/pull/1396 tightly coupled with those requirements as well.